transportable-udfs-avro/src/main/java/com/linkedin/transport/avro/AvroWrapper.java

```diff
@@ -49,8 +49,13 @@ public static StdData createStdData(Object avroData, Schema avroSchema) {
         return new AvroLong((Long) avroData);
       case BOOLEAN:
         return new AvroBoolean((Boolean) avroData);
-      case STRING:
-        return new AvroString((Utf8) avroData);
+      case STRING: {
+        if (avroData instanceof Utf8) {
+          return new AvroString((Utf8) avroData);
+        } else if (avroData instanceof String) {
+          return new AvroString(new Utf8((String) avroData));
```

Comment:
This is fall through case statement where if avroData is neither String nor Utf8, it will continue with the FLOAT label.

BTW, can you add a check before each case statement to make sure Java type of Object match with Avro schema type. Example:
- AvroData should be Integer if the avroSchema type is INT
- AvroData should be String or Utf8 if the avroSchema type is STRING

If not match, we can through our own exception with an explicit message rather than relying on Java type cast handling.

Reply:
Thanks, good catch on switch fall through. Throwing an exception if we reach the end of the STRING case.

Also, I added additional checks for the other data types with explicit error messages.